crates/bevy_state/src/state_scoped.rs

```diff
@@ -91,3 +91,74 @@ pub fn clear_state_scoped_entities<S: States>(
         }
     }
 }
+
+/// Entities marked with this component will be removed
```

Comment:
```suggestion
/// Entities marked with this component will be despawned
```
Optional: The docs for `StateScoped` already used the `removed` wording, but this PR does change the wording in the docs for `despawn_entities_on_exit_state` to `despawned`, and IMO that's more clear.

The docs for `DespawnOnExitState` can also be updated to match.